### PR TITLE
feat(tha-583): treasury module

### PR DESCRIPTION
**Pull request by wesgraham**
Adds a treasury module to thala-common. Treasury has very simple methods, `deposit` and `withdraw`
Ticket: Tha-583

### CODE REVIEW

File: sources/treasury.move [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        assert!(!initialized(signer::address_of(deployer)), ERR_INITIALIZED);
+
+        let (_, signer_cap) = account::create_resource_account(deployer, b"thala_common_treasury_resource_account");
+
+        move_to(deployer, TreasuryAccountCapability { signer_cap });
+    }
+
+    /// Initialize Fee support for CoinType
+    public fun initialize_collateral<CoinType>(deployer: &signer) acquires TreasuryAccountCapability {
+        assert!(initialized(signer::address_of(deployer)), ERR_UNINITIALIZED);
```
Comment (wesgraham) [inline sources/treasury.move:39]: note: are we interested in having the treasury module use the manager module at all? or will these be separate?

**Comment by wesgraham** [comment]
Note: blocked until further discussion on resource account design is completed

**Comment by hamdiallam** [comment]
Is this suppose to replace `fees.move` that exists in all the repos?

File: sources/treasury.move [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        assert!(!initialized_collateral<CoinType>(signer::address_of(deployer)), ERR_INITIALIZED);
+        coin::register<CoinType>(&resource_account_signer(signer::address_of(deployer)));
+    }
+
+    /// Absorb collateral fee. Deposits fee in the resource account CoinStore
+    public entry fun deposit<CoinType>(deployer: address, coin: Coin<CoinType>)
```
Comment (hamdiallam) [inline sources/treasury.move:45]: To avoid having this be called by anyone, should we have this take signer instead of address?

**Review by hamdiallam — COMMENTED** [review]
We could probably just get rid of `initialize` and `initialize_collateral`? I feel like when we deposit, we can just register the coin if it's not registered


Also let's rename `deployer` to `account` or `resource_account`. This module isn't getting deployed

**Comment by hamdiallam** [comment]
@ThalaLabs/engineering 

Given the new structure of singular resource accounts, i'd vote for not having this wrapper anymore and simply interfacing with the `coin` module directly in our repos without a separate fees module. thoughts?

**Comment by wjw12** [comment]
> cc @ThalaLabs/engineering
> 
> Given the new structure of singular resource accounts, i'd vote for not having this wrapper anymore and simply interfacing with the `coin` module directly in our repos without a separate fees module. thoughts?

sounds good

**Pull request closed without merging**